admin/server/reports.go

```diff
@@ -488,6 +489,15 @@ func (s *Server) yamlForManagedReport(opts *adminv1.ReportOptions, ownerUserID s
 	res.Annotations.AdminNonce = time.Now().Format(time.RFC3339Nano)
 	res.Annotations.WebOpenPath = opts.WebOpenPath
 	res.Annotations.WebOpenState = opts.WebOpenState
+	if opts.MetricsViewName != nil || opts.ExploreName != nil {
+		if opts.MetricsViewName == nil || opts.ExploreName == nil {
+			return nil, errors.New("both metricsViewName and exploreName must be provided")
+		}
+		res.Annotations.MetricsViewName = *opts.MetricsViewName
+		res.Annotations.ExploreName = *opts.ExploreName
+	} else if opts.CanvasName != nil {
+		res.Annotations.CanvasName = *opts.CanvasName
+	}
```

Comment:
1. Can it just set all of them and leverage `omitempty` in the YAML serialization?
2. Can we just call these `Explore` and `Canvas` (i.e. omit the `Name` suffix). I think it is clear from context that they are names.
3. Can we avoid including the metrics view here?
    - In the UI they just need the explore name
    - In the report reconciler, it can lookup the explore and find its underlying metrics view